src/bench/mempool_eviction.cpp

```diff
@@ -21,116 +21,94 @@ static void AddTx(const CTransactionRef& tx, const CAmount& nFee, CTxMemPool& po
         spendsCoinbase, sigOpCost, lp));
 }
 
-// Right now this is only testing eviction performance in an extremely small
-// mempool. Code needs to be written to generate a much wider variety of
-// unique transactions for a more meaningful performance measurement.
-static void MempoolEviction(benchmark::Bench& bench)
+void add_parents_child(std::vector<CTransactionRef>& txns, size_t package_size, size_t set_num)
+{
+    // N+1 N-input-N-output txns, in a parents-and-child package
+    // Where each tx takes an input from all prior txns
+    const size_t num_puts = std::max(package_size - 1, (size_t)1);
+    const size_t num_txns = package_size;
+    for (size_t txn_index=0; txn_index<num_txns; txn_index++) {
+        CMutableTransaction tx = CMutableTransaction();
+        tx.vin.resize(num_puts);
+        tx.vout.resize(num_puts);
+        for (size_t put_index=0; put_index<num_puts; put_index++) {
+            tx.vin[put_index].scriptSig = CScript() << CScriptNum(txn_index);
+            tx.vin[put_index].scriptWitness.stack.push_back({1});
+            tx.vout[put_index].scriptPubKey = CScript() << CScriptNum(set_num); // Ensures unique txid with set_num
+            tx.vout[put_index].nValue = txn_index;
+
+            if (put_index < txn_index) {
+                // We spend put_index's txn's in the next available slot for each previous transaction
+                assert(put_index + txn_index >= 1);
+                tx.vin[put_index].prevout = COutPoint(txns[put_index]->GetHash(), put_index + txn_index - 1);
```

Comment:
Wait, if you're using `txns[put_index]` *and* you're using the same `txns` reference each time you call `add_parents_child`, aren't you having all subsequent packages spend outputs from txns[0:24] too? That means all these packages conflict with each other.

You could fix this by adding `set_num * package_size` or something to the index. But what would probably make a better interface is if `add_parents_child` returns the list of transactions it creates, and you append them to your larger list of packages.

Reply:
Hmmm, traced out a package size of 4, looks like I'm referencing non-existent outputs, not re-using them(maybe both :) ).

Clearly this is confusing and broken either way.

To take a step back, after doing this I realized https://github.com/bitcoin/bitcoin/blob/master/src/bench/mempool_stress.cpp exists. I'm fine if this doesn't get merged if it doesn't add anything new? Helped me learn about the benchmarks if nothing else. Thoughts?

Reply:
I could change the PR to remove this silly bench.. :P